### PR TITLE
Introduce more variables to find CUDA with

### PR DESCRIPTION
**Pull request by teonnik**
- Do not rely on `enable_language(CUDA)` which was removed
- Introduce/Document alternative commonly used environment/CMake variables for finding CUDA libraries

### CODE REVIEW

**Review by haampie — APPROVED** [review]
LGTM, I'm just using `-DCUDALIBS_ROOT=/usr/local/cuda/targets/x86_64-linux` for the ubuntu:18.04 docker image now.

File: cmake/FindCUDALIBS.cmake [hunk 1]
```diff
@@ -21,13 +23,21 @@
...
 find_path(CUDA_CUBLAS_INCLUDE cublas_v2.h
-          PATHS ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
+          PATHS ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}
+                $ENV{CUDA_HOME}
+          PATH_SUFFIXES include)
 find_library(CUDA_CUDART_LIB cudart
-             PATHS ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES})
+             PATHS ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES}
+                   $ENV{CUDA_HOME}
+             PATH_SUFFIXES lib64)
```
Comment (albestro) [inline cmake/FindCUDALIBS.cmake:36]: Is it always `lib64`?

**Review by albestro — APPROVED** [review]
LGTM

File: cmake/FindCUDALIBS.cmake (hunk 1 above)
Comment (msimberg) [reply cmake/FindCUDALIBS.cmake:36]: That's in addition to `lib`, I think. @teonnik please correct me if I'm wrong. There's also this https://cmake.org/cmake/help/v3.12/prop_gbl/FIND_LIBRARY_USE_LIB64_PATHS.html#prop_gbl:FIND_LIBRARY_USE_LIB64_PATHS which I think would work as an alternative.

Comment (teonnik) [reply cmake/FindCUDALIBS.cmake:36]: >  That's in addition to lib

Yes, I think so too. The docs, point 1. seems to agree : https://cmake.org/cmake/help/v3.19/command/find_library.html#command:find_library

> here's also this https://cmake.org/cmake/help/v3.12/prop_gbl/FIND_LIBRARY_USE_LIB64_PATHS.html#prop_gbl:FIND_LIBRARY_USE_LIB64_PATHS which I think would work as an alternative.

I wasn't aware of that one. :)

**Pull request merged**